### PR TITLE
Content$delete_tag removes that tag from a content item

### LINKED ISSUE #194: Content$tag_delete() deletes a tag altogether instead of removing a tag from a Content object
* `Content$tag_delete()` says it "Removes a tag for this content."
* My observation is that it not only removes the tag from the content, it also deletes the tag entirely
* This is misleading and should be fixed. There should be separate methods for removing a tag from a content object, vs deleting a tag altogether.

https://pkgs.rstudio.com/connectapi/reference/Content.html#method-tag-delete-

```r
# insert reprex here
```

### PR DESCRIPTION
**Pull request by aronatkins**
fixes #194

### CODE REVIEW

**Review by colearendt — APPROVED** [review]

**Review by colearendt — APPROVED** [review]
Looks great! Just one little nit-pick. I tested this interactively - once we get the dev environment more stable (hard to run on an M1 right now), we should definitely add tag deletion to the integration tests 😄

File: R/content.R [hunk 1]
```diff
@@ -141,9 +141,9 @@ Content <- R6::R6Class(
     },
     #' @description Remove a tag for this content.
     #' @param id The tag identifier.
-    tag_delete = function(id) {
+    tag_delete = function(tag_id) {
       # note that deleting the parent tag deletes all children
```
[the remark this answers is not in the document]
Comment (colearendt) [reply R/content.R:145]: Oh right! I was reading it as what it was doing previously (deleting the parent tag / children). You're right, this does remove the children from the content. 🤦 Sorry about that! Good to merge 😄

**Comment by fh-kpikhart** [comment]
I might be reading this wrong, but the way I interpret this code change, permanently deleting tags is now a method belonging to the _Content_ class, while removing tags from content is a method belonging to the _Connect_ class. That is the opposite of what I would expect. Just FYI.

**Comment by aronatkins** [comment]
@fh-kpikhart I think that you are misinterpreting the change.

The `Content` class has many wrappers which delegate to the `Connect` class.

The `Content$tag_delete(tag_id)` function removes the tag for a single content item. Behind the scenes, that calls `Connect$remove_content_tag(content_id, tag_id)`. 

The `Connect$remove_content_tag(content_id, tag_id)` function also removes the tag for a single content item, but requires that the caller pass both the content and tag identifiers.

The `Connect$tag_delete(id)` function, in contrast, removes a tag entirely from the system. That function is unchanged.

Prior to my change, the `Content$tag_delete(tag_id)` function called `Connect$tag_delete(id)`, which was incorrect.

**Comment by fh-kpikhart** [comment]
Got it. Thank you!

**Pull request merged**